sdk/src/test/scala/io/horizen/account/sc2sc/AbstractCrossChainRedeemMessageProcessorTest.scala

```diff
@@ -0,0 +1,233 @@
+    when(mockStateView.getAccountStorage(contractAddress, ccMsgHash)).thenReturn(Array.emptyByteArray)
+
+    when(scTxCommProvider.doesScTxCommitmentTreeRootHashExist(scCommitmentTreeRoot, mockStateView)).thenReturn(true)
+    when(scTxCommProvider.doesScTxCommitmentTreeRootHashExist(nextScCommitmentTreeRoot, mockStateView)).thenReturn(true)
+
+    when(sc2scCircuitMock.getCrossChainMessageHash(any())).thenReturn(crossChainMessageHash)
+    when(sc2scCircuitMock.verifyRedeemProof(ArgumentMatchers.eq(crossChainMessageHash), any(), any(), any())).thenReturn(false)
+
+    // Act
+    val exception = intercept[ExecutionRevertedException] {
+      withGas(ccRedeemMsgProcessor.process(msg, mockStateView, _, defaultBlockContext))
+    }
+
+    // Assert
+    val expectedMsg = s"Cannot verify this cross-chain message"
+    assertTrue(exception.getCause.getMessage.contains(expectedMsg))
+  }
+
+  @Test
+  def whenAllValidationsPass_throwsNoException(): Unit = {
+    // Arrange
+    val ccRedeemMsgProcessor: AbstractCrossChainRedeemMessageProcessor = new CrossChainRedeemMessageProcessorImpl(networkParamsMock, sc2scCircuitMock, scTxCommProvider)
+    val msg = mock[Message]
+
+    when(networkParamsMock.sidechainId).thenReturn(receiverSidechain)
+
+    when(sc2scCircuitMock.getCrossChainMessageHash(any())).thenReturn(crossChainMessageHash)
+    when(crossChainMessageHash.getValue).thenReturn("someBytes".getBytes)
+    val ccMsgHash = ccRedeemMsgProcessor.getCrossChainMessageFromRedeemKey(crossChainMessageHash.getValue)
+    when(mockStateView.getAccountStorage(contractAddress, ccMsgHash)).thenReturn(Array.emptyByteArray)
+
+    when(scTxCommProvider.doesScTxCommitmentTreeRootHashExist(scCommitmentTreeRoot, mockStateView)).thenReturn(true)
+    when(scTxCommProvider.doesScTxCommitmentTreeRootHashExist(nextScCommitmentTreeRoot, mockStateView)).thenReturn(true)
+
+    when(sc2scCircuitMock.getCrossChainMessageHash(any())).thenReturn(crossChainMessageHash)
+    when(sc2scCircuitMock.verifyRedeemProof(ArgumentMatchers.eq(crossChainMessageHash), any(), any(), any())).thenReturn(true)
+
+    // Act & Assert
+    try withGas(ccRedeemMsgProcessor.process(msg, mockStateView, _, defaultBlockContext))
+    catch {
+      case _: Exception =>
+        fail("Test failed unexpectedly")
+    }
+  }
+}
+
+class CrossChainRedeemMessageProcessorImpl(networkParams: NetworkParams, sc2scCircuit: Sc2scCircuit, scTxCommTreeRootProvider: ScTxCommitmentTreeRootHashMessageProvider)
+  extends AbstractCrossChainRedeemMessageProcessor(networkParams, sc2scCircuit, scTxCommTreeRootProvider) {
+
+  override val contractAddress: Address = CrossChainRedeemMessageProcessorImpl.contractAddress
+  override val contractCode: Array[Byte] = CrossChainRedeemMessageProcessorImpl.contractCode
+
+  override protected def getAccountCrossChainRedeemMessageFromMessage(msg: Message): AccountCrossChainRedeemMessage = {
+    val accountCrossChainMessage = AccountCrossChainMessage(
+      messageType = 1,
```

Comment:
What is the meaning of the `messageType` field? Is '1' the only supported value? Consider using an enum for instance in the `AccountCrossChainMessage` class.